### PR TITLE
feat/ integrate new `APIThrottler` into `binance` and `binance_us` connector

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Integrates the new `APIThrottler` with the Binance connector

**Tests performed by the developer**:
Added Rate Limits constants
Refactors usage of API URLs

**Tips for QA testing**:
Test that `binance` and `binance_us` connector works as intended.
Note: Do modify the `rate_limits_share_pct` config to forcibly hit the API rate limits.

### CODE REVIEW

**Comment by RC-13** [comment]
@CrimsonJacket  I keep getting  server rejected WebSocket connection: HTTP 400 on both Binance.com and Binance_US, it doesn't matter what refresh rate is use. 
image

```python
2021-09-07 13:50:11,916 - 11599 - hummingbot.connector.exchange.binance.binance_api_user_stream_data_source - ERROR - Unexpected error while maintaining the user event listen key. Retrying after 5 seconds... Error: server rejected WebSocket connection: HTTP 400
Traceback (most recent call last):
  File "/Users/clintgogoc/Desktop/github/binance-throttler/hummingbot/connector/exchange/binance/binance_api_user_stream_data_source.py", line 155, in listen_for_user_stream
    self._ws: websockets.WebSocketClientProtocol = await self.get_ws_connection()
  File "/Users/clintgogoc/Desktop/github/binance-throttler/hummingbot/connector/exchange/binance/binance_api_user_stream_data_source.py", line 118, in get_ws_connection
    ws = await websockets.connect(stream_url)
  File "/Users/clintgogoc/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/client.py", line 542, in __await_impl__
    await protocol.handshake(
  File "/Users/clintgogoc/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/client.py", line 296, in handshake
    raise InvalidStatusCode(status_code)
websockets.exceptions.InvalidStatusCode: server rejected WebSocket connection: HTTP 400
```

File: hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py [hunk 1]
```diff
@@ -44,97 +45,113 @@ def logger(cls) -> HummingbotLogger:
...
-            snapshot_timestamp: float = time.time()
-            snapshot_msg: OrderBookMessage = BinanceOrderBook.snapshot_message_from_exchange(
-                snapshot,
-                snapshot_timestamp,
-                metadata={"trading_pair": trading_pair}
-            )
-            order_book = self.order_book_create_function()
-            order_book.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
-            return order_book
+        snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair, 1000, self._domain)
```
Comment (petioptrv) [inline hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py:145]: In this case, we're able to pass the throttler of the instance to `get_snapshot`

File: hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py [hunk 2]
```diff
@@ -201,30 +216,27 @@ async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, outp
...
-                        except Exception:
-                            self.logger().error("Unexpected error.", exc_info=True)
-                            await asyncio.sleep(5.0)
-                    this_hour: pd.Timestamp = pd.Timestamp.utcnow().replace(minute=0, second=0, microsecond=0)
-                    next_hour: pd.Timestamp = this_hour + pd.Timedelta(hours=1)
-                    delta: float = next_hour.timestamp() - time.time()
-                    await asyncio.sleep(delta)
+                for trading_pair in self._trading_pairs:
+                    try:
+                        snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair, domain=self._domain)
```
Comment (petioptrv) [inline hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py:221]: In this case, we're able to pass the throttler of the instance to `get_snapshot`

**Review by petioptrv — COMMENTED** [review]
Only one comment, the rest LGTM

**Comment by PtrckM** [comment]
> ```python
> 2021-09-07 13:50:11,916 - 11599 - hummingbot.connector.exchange.binance.binance_api_user_stream_data_source - ERROR - Unexpected error while maintaining the user event listen key. Retrying after 5 seconds... Error: server rejected WebSocket connection: HTTP 400
> Traceback (most recent call last):
>   File "/Users/clintgogoc/Desktop/github/binance-throttler/hummingbot/connector/exchange/binance/binance_api_user_stream_data_source.py", line 155, in listen_for_user_stream
>     self._ws: websockets.WebSocketClientProtocol = await self.get_ws_connection()
>   File "/Users/clintgogoc/Desktop/github/binance-throttler/hummingbot/connector/exchange/binance/binance_api_user_stream_data_source.py", line 118, in get_ws_connection
>     ws = await websockets.connect(stream_url)
>   File "/Users/clintgogoc/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/client.py", line 542, in __await_impl__
>     await protocol.handshake(
>   File "/Users/clintgogoc/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/client.py", line 296, in handshake
>     raise InvalidStatusCode(status_code)
> websockets.exceptions.InvalidStatusCode: server rejected WebSocket connection: HTTP 400
> ```
- confirmed this issue is fix

---

- getting this error while running at `rate_limits_share_pct 90` and/or 100% (randomly)

random11
random1

- when it appears then try to stop the bot or exit it will fail to cancel the orders (some orders will get stuck on exchange but not all)
image

**Review by petioptrv — APPROVED** [review]
The changes LGTM

**Review by PtrckM — APPROVED** [review]
Confirm issues found are fixed. Approving in behalf of QA

**Pull request merged**